panther_manager/include/panther_manager/plugins/ros_service_node.hpp

```diff
@@ -13,53 +18,57 @@ namespace panther_manager
 template <class ServiceT>
 class RosServiceNode : public BT::SyncActionNode
 {
-protected:
-  RosServiceNode(
-    std::shared_ptr<ros::NodeHandle> & nh, const std::string & name, const BT::NodeConfig & conf)
-  : BT::SyncActionNode(name, conf), nh_(nh)
+public:
+  explicit RosServiceNode(const std::string & name, const BT::NodeConfig & conf)
+  : BT::SyncActionNode(name, conf)
   {
-    if (!getInput<std::string>("service_name", srv_name_) || srv_name_ == "") {
-      throw BT::RuntimeError("[", name, "] Failed to get input [service_name]");
-    }
-
-    unsigned srv_timeout_ms;
-    if (!getInput<unsigned>("timeout", srv_timeout_ms)) {
-      throw BT::RuntimeError("[", name, "] Failed to get input [timeout]");
-    }
-    srv_timeout_ = ros::Duration(static_cast<double>(srv_timeout_ms) * 1e-3);
-
+    nh_ = std::move(config().blackboard->get<std::shared_ptr<ros::NodeHandle>>("nh"));
```

Comment:
```suggestion
    nh_ = std::make_shared(config().blackboard->get<std::shared_ptr<ros::NodeHandle>>("nh"));
```
This should be copying constructor. Not moving the pointer

Reply:
After discussion let's leave it as it is